tp_framework/core/utils.py

```diff
@@ -40,14 +42,27 @@ def list_pattern_instances_by_pattern_id(language: str, pattern_id: int, tp_lib_
         raise ee
 
 
+def get_tpi_id_from_jsonpath(jp: Path) -> int:
+    return get_id_from_name(jp.parent.name)
+
 def get_pattern_dir_from_id(pattern_id: int, language: str, tp_lib_dir: Path) -> Path:
     tp_lib_dir_lang_dir: Path = tp_lib_dir / language
     if tp_lib_dir_lang_dir.is_dir():
-        return list(filter(lambda p: int(p.name.split("_")[0]) == pattern_id, tp_lib_dir_lang_dir.iterdir()))[0]
+        return list(filter(lambda p: get_id_from_name(p.name) == pattern_id, list_dirs_only(tp_lib_dir_lang_dir)))[0]
     else:
         raise PatternDoesNotExists(pattern_id)
 
 
+def get_instance_dir_from_id(instance_id: int, pattern_dir: Path) -> Path:
+    if pattern_dir.is_dir():
+        return get_instance_dir_from_list(instance_id, list_dirs_only(pattern_dir))
+    else:
+        raise PatternFolderNotFound()
+
+
+def get_instance_dir_from_list(instance_id: int, l_pattern_dir: list[Path]):
+    return list(filter(lambda tpi_dir: get_id_from_name(tpi_dir.name) == instance_id, l_pattern_dir))[0]

```

Comment:
```suggestion
    instance_with_id = list(filter(lambda tpi_dir: get_id_from_name(tpi_dir.name) == instance_id, l_pattern_dir))
    if not instance_with_id:
        raise InstanceDoesNotExists()
    return instance_with_id[0]
```
This suggestion follows the same argumentation as the comment above. If there is no instance with the wanted ID, the current version throws a `KeyError`.  This suggestion would throw an `InstanceDoesNotExists()` error.